L2/gamma_testnet_fdg_upgrade.md

```diff
@@ -0,0 +1,95 @@
+0. Prepare a pre-funded `$prefunded_pk` to be used for deploying contracts below.
+1. Upgrade ASR：
+    ```bash
+    pushd optimism/packages/contracts-bedrock
+    # deploy UpgradeAnchorStateRegistry contract
+    forge create scripts/deploy/UpgradeAnchorStateRegistry.s.sol:UpgradeAnchorStateRegistry \
+            --broadcast \
+            --private-key $prefunded_pk \
+            --rpc-url $L1_RPC_URL
+    popd
+    cast calldata 'run(address,address,address,address,address,uint32,bytes32,uint256)' \
+        $DISPUTE_GAME_FACTORY_PROXY_ADDRESS $OP_PROXY_ADMIN_ADDRESS \
+        $ANCHOR_STATE_REGISTRY_PROXY_ADDRESS $SUPERCHAIN_CONFIG_PROXY_ADDRESS \
+        0x0000000000000000000000000000000000000000 \
+        0 0xa892c858b32ddb0d5c7c5a53690a28c3163a4ee21c06f7b6000c3db6a05db108 0
+    Delegate call the UpgradeAnchorStateRegistry contract with above calldata from Safe
+    ```
+    1.  fetch genesis output root：`0xa892c858b32ddb0d5c7c5a53690a28c3163a4ee21c06f7b6000c3db6a05db108`
+        ```bash
+        $ curl -X POST -H "Content-Type: application/json" --data  \
+        '{"jsonrpc":"2.0","method":"optimism_outputAtBlock","params":["0x0"],"id":1}' \
+        http://65.109.69.90:8547
+        ```
+2. Run `make reproducible-prestate` to get correct absolute prestate(MIPS64)：`0x031561ec3552bc669bacc5467cc87f51268e764508baa2d3735b5526a1c4f94e`
+3. Attributes from the permissioned FDG can be queried like below：
+```bash
+# cast call $DISPUTE_GAME_FACTORY_PROXY_ADDRESS 'gameImpls(uint32)(address)' 1 -r $L1_RPC_URL
+0x29014B28390e403a0f0885330a97dbeB70C66fBf
+# cast call 0x29014B28390e403a0f0885330a97dbeB70C66fBf "maxGameDepth()(uint256)" -r $L1_RPC_URL
+73
+# cast call 0x29014B28390e403a0f0885330a97dbeB70C66fBf "splitDepth()(uint256)" -r $L1_RPC_URL
+30
+# cast call 0x29014B28390e403a0f0885330a97dbeB70C66fBf "clockExtension()(uint64)" -r $L1_RPC_URL
+10800
+# cast call 0x29014B28390e403a0f0885330a97dbeB70C66fBf "maxClockDuration()(uint64)" -r $L1_RPC_URL
+302400
+# cast call 0x29014B28390e403a0f0885330a97dbeB70C66fBf "vm()(address)" -r $L1_RPC_URL
+0xF027F4A985560fb13324e943edf55ad6F1d15Dc1
+```
+4. Run the command below to deploy a permission-less FDG with correct absolute prestate：
+```bash
+pushd optimism/packages/contracts-bedrock
+forge create src/dispute/FaultDisputeGame.sol:FaultDisputeGame \
+            --broadcast \
+            --private-key $prefunded_pk \
+            --rpc-url $L1_RPC_URL \
+            --constructor-args \
+            0 \ # gameType
+            0x037954296697a98e3a22764cdbfc0820e45219eed5dbf6795160f060b19031bc \ # absolutePrestate
```

Comment:
Should also update to 0x031561ec3552bc669bacc5467cc87f51268e764508baa2d3735b5526a1c4f94e?